examples/cityscapes/leap_binder.py

```diff
@@ -0,0 +1,308 @@
+    """
+    data = data.data
+    cloud_path = data['gt_bbx_path'][idx%data["real_size"]]
+    fpath = _download(cloud_path)
+    with open(fpath, 'r') as file:
+        json_data = json.load(file)
+    bounding_boxes = extract_bounding_boxes_from_instance_segmentation_polygons(json_data)
+    return bounding_boxes
+
+# ----------------------------------------------------------metadata----------------------------------------------------
+
+def number_of_bb(index: int, subset: PreprocessResponse) -> int:
+    bbs = np.array(ground_truth_bbox(index, subset))
+    number_of_bb = np.count_nonzero(bbs[..., -1] != BACKGROUND_LABEL)
+    return number_of_bb
+
+def instances_num(index: int, subset: PreprocessResponse) -> float:
+    bbs = np.array(ground_truth_bbox(index, subset))
+    valid_bbs = bbs[bbs[..., -1] != BACKGROUND_LABEL]
+    return float(valid_bbs.shape[0])
+
+def avg_bb_aspect_ratio(index: int, subset: PreprocessResponse) -> float:
+    bbs = np.array(ground_truth_bbox(index, subset))
+    valid_bbs = bbs[bbs[..., -1] != BACKGROUND_LABEL]
+    assert ((valid_bbs[:, 3] > 0).all())
+    aspect_ratios = valid_bbs[:, 2] / valid_bbs[:, 3]
+    return aspect_ratios.mean()
+
+def avg_bb_area_metadata(index: int, subset: PreprocessResponse) -> float:
+    bbs = np.array(ground_truth_bbox(index, subset)) # x,y,w,h
+    valid_bbs = bbs[bbs[..., -1] != BACKGROUND_LABEL]
+    areas = valid_bbs[:, 2] * valid_bbs[:, 3]
+    return areas.mean()
+
+def label_instances_num(class_label: str) -> Callable[[int, PreprocessResponse], float]:
+    def func(index: int, subset: PreprocessResponse) -> float:
+        bbs = np.array(ground_truth_bbox(index, subset))
+        label = CATEGORIES.index(class_label)
+        valid_bbs = bbs[bbs[..., -1] == label]
+        return float(valid_bbs.shape[0])
+
+    func.__name__ = f'metadata_{class_label}_instances_count'
+    return func
+
+def is_class_exist_gen(class_id: int) -> Callable[[int, PreprocessResponse], float]:
+    def func(index: int, subset: PreprocessResponse):
+        bbs = np.array(ground_truth_bbox(index, subset))
+        is_i_exist = (bbs[..., -1] == class_id).any()
+        return is_i_exist
+
+    func.__name__ = f'metadata_{class_id}_instances_count'
+    return func
+
+def is_class_exist_veg_and_building(class_id_veg: int, class_id_building) -> Callable[[int, PreprocessResponse], float]:
+    def func(index: int, subset: PreprocessResponse):
+        bbs = np.array(ground_truth_bbox(index, subset))
+        is_veg_exist = (bbs[..., -1] == class_id_veg).any()
+        is_building_exist = (bbs[..., -1] == class_id_building).any()
+        return is_veg_exist and is_building_exist
+
+    func.__name__ = f'metadata_class_veg_and_class_building_instances_count'
+    return func
+
+
+def get_class_mean_iou(class_id: int = None) -> Callable[[Tensor, Tensor], Tensor]:
+
+    def class_mean_iou(y_true: tf.Tensor, y_pred: tf.Tensor) -> tf.Tensor:
```

Comment:
If possible - remove this to some other file to make this file shorter

Reply:
moved to metrices